src/components/card/card-dialog/CardDialog.js

```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import { TextField } from '@mui/material';
+import { v4 as uuidv4 } from 'uuid';
+import PropTypes from 'prop-types';
+
+function CardDialog({ params }) {
+  const [open, setOpen] = useState(false);
+  const [cardTitle, setTitle] = useState(null);
+  const [cardBody, setBody] = useState(null);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const clearState = () => {
+    setTitle(null);
+    setBody(null);
+  };
+
+  function isCardValid() {
+    const isValid = !cardTitle || !cardBody || !cardTitle.trim().length || !cardBody.trim().length;
+    return !isValid;
+  }
```

Comment:
instead of that I would use util function isBlank and move it outside the component

```
/**
 * Checks that provided string is blank
 * @param target {string}
 */
function isBlank(target) {
    return !target || !target.trim();
}
```